maintenance/wikia/TemplateClassificationRecognizedMetric.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+require_once __DIR__ . '/../Maintenance.php';
+
+/**
+ * Maintenance script counts current number of recognized templates in content namespaces
+ *
+ * @author Kamil Koterba <kamil@wikia-inc.com>
+ * @ingroup Maintenance
+ */
+class TemplateClassificationRecognizedMetricMaintenance extends Maintenance {
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->mDescription = 'Maintenance script counts current number of recognized templates in content namespaces';
+	}
+
+	/**
+	 * Script entry point
+	 */
+	public function execute() {
+		global $wgContentNamespaces, $wgDWStatsDB;
+
+		$statsDBRead = wfGetDB( DB_SLAVE, [], $wgDWStatsDB );
+		$wamTop500 = $this->getWamTop500( $statsDBRead, ( new \WikiaSQL ) );
+
+		$top500CNRecognizedCount = 0;
+		$top500CNAllCount = 0;
+
+		foreach ( $wamTop500 as $wikia ) {
```

Comment:
It's a design decision so there were some trade-offs considered. I would recommend surfacing them here. I see at least two approaches; The one proposed here is executed in sequence; the second would be moving WAM top 500 outside and running the calculation in context of one wiki would allow for parallel execution.

It may be worthwhile to add a comment about this :)